### ISSUE TITLE
Ultraloq Pro Z-wave

### ISSUE THREAD

**Issue by JPDMaker** [AlCalzone/node-zwave-js#5445]
### Is your problem within Home Assistant (Core or Z-Wave JS Integration)?

NO, my problem is NOT within Home Assistant or the ZWave JS integration

### Is your problem within Z-Wave JS UI (formerly ZwaveJS2MQTT)?

NO, my problem is NOT within Z-Wave JS UI

### Checklist

- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there.

### Describe the bug

[77-98-0-latchStatus] Current status of the latch is always reporting "OPEN"

U-tec support said the issue could be on the z wave hub side.
Thread of others reporting the same issue: https://community.u-tec.com/t/expand-and-fix-z-wave-capabilities/502/18

### Device information

Manufacturer: u-tec
Model name: U Bolt Pro Z wave 
Node ID in your network: <077>

### How are you using `node-zwave-js`?

- [X] `zwave-js-ui` (formerly `zwavejs2mqtt`) Docker image (latest)
- [ ] `zwave-js-ui` (formerly `zwavejs2mqtt`) Docker image (dev)
- [ ] `zwave-js-ui` (formerly `zwavejs2mqtt`) Docker manually built _(please specify branches)_
- [ ] `ioBroker.zwave2` adapter _(please specify version)_
- [X] `HomeAssistant zwave_js` integration  _(please specify version)_
- [ ] `pkg`
- [ ] `node-red-contrib-zwave-js` _(please specify version, double click node to find out)_
- [ ] Manually built from GitHub _(please specify branch)_
- [ ] Other _(please describe)_

### Which branches or versions?

zwave-js-ui: 8.6.3.c12b1c6
zwave-js: 10.3.1

### Did you change anything?

no

### Did this work before?

Don't know, this is a new device

### Attach Driver Logfile

[zwavejs_current.log](https://github.com/zwave-js/node-zwave-js/files/10701445/zwavejs_current.log)

**Comment by efaden** [comment]
Having same issue, also the door status is always open.  Doesn't seem to report the state of the sensors correctly.  According to their zwave documentation both should be on the reports.  Different parts of the door condition

**Comment by efaden** [comment]
![Screenshot_20230211-102053](https://user-images.githubusercontent.com/12599300/218266463-347d685b-ce29-4f4d-8139-9d0815660ace.png)

**Comment by AlCalzone** [comment]
I'll try on Monday

**Comment by efaden** [comment]
Thanks.  There really should be two entities for the pro if the sensor is present.  One for the lock, and one for the door sensor

**Comment by AlCalzone** [comment]
That's a bug in the device firmware, at least in v1.1.3 which my lock is on.
When being queried for the supported door components, it has the byte set to 6:
![grafik](https://user-images.githubusercontent.com/17641229/218426601-0f3025b0-7daf-42fb-a08a-bde7da005887.png)

which means Bolt and Latch states are supported, Door state isn't:
![grafik](https://user-images.githubusercontent.com/17641229/218426952-491fdaaf-7db0-4eeb-bc75-630d01f8736f.png)

Given that the manual states that Bolt and Door are supported, this seems like a bug.

**Comment by AlCalzone** [comment]
Despite that, I can't get the lock to report any change in door state. It always reports door open.

**Comment by efaden** [comment]
Well that's not awesome.  I send an email to them about this issue with this thread... Maybe they can fix it.  Would be great to be able to get door status

**Comment by tekkenly** [comment]
Hi @AlCalzone ,

I wonder if we can put some special code in to make a fix for this device and this version firmware? the outdoorhandle value is clearly matching the door sensor state.

Or, if you can give me a quick hint where the related code is, I can try to make a change, then you can choose if you want to merge it in.

Thanks.

**Comment by AlCalzone** [comment]
We don't hardcode such one-time device-specific behavior.

**Comment by AlCalzone** [comment]
I've heard that U-Tec finally managed to reproduce the issue, but no idea if and when this will be fixed.

**Comment by johntdyer** [comment]
@AlCalzone  do you have a link to their issue I can follow ?

**Comment by AlCalzone** [comment]
Nope, that was all back-channel information. Haven't head anything since.

**Comment by TheOneOgre** [comment]
I would like to add that they have seemingly made no progress on fixing this issue and I have discovered a second issue and it seems others have as well.

When the lock is triggered to lock, even if the locking process fails, say the door isn't closed all the way or something happens to make it not engage the deadbolt, the lock will beep and error out but the reported status to z-wave is that the lock was successful. This is dangerous as even though you should always manually check for security, it should report some kind of error or at least fallback to the previous state. Kind of disappointed in Ultraloq on the development of this.

**Comment by efaden** [comment]
Yeah, the lock is good, but if they don't actually fix bugs then it's
really not a great choice

**Comment by TheOneOgre** [comment]
> Yeah, the lock is good, but if they don't actually fix bugs then it's really not a great choice

I completely agree. The firmware version of my lock is 02.26.0022 which, from how it reads, is from February of 2022, so no firmware updates in almost a year and a half. For something that's supposed to be a "Work With Home Assistant" product, that's a pretty unacceptable lack of updates especially when there are glaring bugs.

**Comment by balloob** [comment]
We have decided to terminate the Works with Home Assistant relationship with Ultraloq. We had reported these issues with them and were discussing solutions. However, they ceased communication from their end and we have received no further updates.

**Comment by TheOneOgre** [comment]
> We have decided to terminate the Works with Home Assistant relationship with Ultraloq. We had reported these issues with them and were discussing solutions. However, they ceased communication from their end and have received no further updates.

Good to see some action being taken about this. Thanks for keeping Home Assistant great!

**Comment by AlCalzone** [comment]
I'm afraid there's nothing to be done in this project, since the functionality is just broken.

**Comment by holblin** [comment]
Could we update https://www.home-assistant.io/integrations/ultraloq/ ?

**Comment by b-uwe** [comment]
I'll do that on Tuesday!

**Comment by b-uwe** [comment]
@TheOneOgre @holblin @AlCalzone https://github.com/home-assistant/home-assistant.io/pull/28166

**Comment by jack0li** [comment]
This problem has been solved in the new firmware, which can be upgraded online through Home assistant.

**Comment by God-TM** [comment]
Will it be added back into the WWHA partner program?

**Comment by jack0li** [comment]
Yes, we do our best to add it back.

Thanks!



***@***.***
 
From: God-TM
Date: 2023-08-23 08:51
To: zwave-js/node-zwave-js
CC: jack; Comment
Subject: Re: [zwave-js/node-zwave-js] Ultraloq Pro Z-wave (Issue #5445)
Will it be added back into the WWHA partner program?

**Comment by TheOneOgre** [comment]
> Will it be added back into the WWHA partner program?

I wouldn't think it would or should be added back. Ignoring HA devs, not providing timely updates, not notifying users of issues, and overall not being easy to work with, is an easy disqualifier for being in a partner program. I think the biggest thing was the untimely manner in which they handled this and the complete lack of communication.

Also, I haven't tested it yet but, can anyone confirm if the "false lock" bug is still occurring where if the lock is told to lock, but fails due to the door being slightly open or something, the state is still reported as "Locked"? Was a HUGE oversight before if you relied on that status for your door being locked.

**Comment by brettmiller** [comment]
> Also, I haven't tested it yet but, can anyone confirm if the "false lock" bug is still occurring where if the lock is told to lock, but fails due to the door being slightly open or something, the state is still reported as "Locked"? Was a HUGE oversight before if you relied on that status for your door being locked.

Tested and the "false lock" bug still exists.

**Comment by b-uwe** [comment]
> > Will it be added back into the WWHA partner program?

> I wouldn't think it would or should be added back. Ignoring HA devs, not providing timely updates, not notifying users of issues, and overall not being easy to work with, is an easy disqualifier for being in a partner program.

Things have changed drastically in the last few weeks! Please feel free to put that to a test! I' doing exactly so with the last statement, the "false lock" bug which I expect to be worked on very soon.
But that's why, yes, we plan to add them back to the WWHA partner program!

**Comment by jack0li** [comment]
> > Also, I haven't tested it yet but, can anyone confirm if the "false lock" bug is still occurring where if the lock is told to lock, but fails due to the door being slightly open or something, the state is still reported as "Locked"? Was a HUGE oversight before if you relied on that status for your door being locked.
> 
> Tested and the "false lock" bug still exists.

Thanks for Testing.  I will check it ASAP.

**Comment by God-TM** [comment]
I just tested it. Perhaps I'm not understanding what's wrong here, but when I open and close the door, the sensor labled "Current status of the door" goes from Open to Closed and vice-versa accurately reflecting the state of the door. I can tell it to lock in either state, and the state is still either Open or Closed depending on if it's actually open or closed...

**Comment by jack0li** [comment]
I have tested it, the door status is correct depending the open or close operation.  
Please put the door magnet as close as possible to the lock, the farther the magnet is, the weaker the magnetic is.
Can you show me your test environment picture  include the lock and door magnet ？

**Comment by God-TM** [comment]
I used to have a Gen 2 August Pro there, so I left the magnet from that and used it with the Ultraloq Pro. So far it's been working without any issues.
![IMG_8812](https://github.com/zwave-js/node-zwave-js/assets/133412022/c74f6238-36a4-44e2-b54b-f1880acaa2f9)
![IMG_8813](https://github.com/zwave-js/node-zwave-js/assets/133412022/92ac4661-4ef8-455f-8bb9-377a4cafcd05)
![IMG_8814](https://github.com/zwave-js/node-zwave-js/assets/133412022/91f80705-9cd8-4e4b-9ce6-da4e73b6d3f7)
![IMG_8815](https://github.com/zwave-js/node-zwave-js/assets/133412022/1d19a09b-fcfd-43e7-9862-e8b945df1a57)
![IMG_8816](https://github.com/zwave-js/node-zwave-js/assets/133412022/0109c94f-7089-4f78-b0f2-a03751c67d6a)

Please let me know if I can take any better measurements or angles that might be helpful for you.

I did test the scenario where you close it mostly but leave it a bit misaligned. If I tell it to lock, it technically goes into a lock status (and I see I can unlock the door). But the sensor is still showing the door is open as well. I don't believe this is any different than what the August Pro would do.

**Comment by jack0li** [comment]
Thanks for your info.
1.  Can you gvie your lock Serial Number （SN：xxxxx）,inside of the battery cover ?
2. The magnet detection chip in the lock  and the detection algorithm are different from August, we don't  do compatibility test with August magnet.  Can you install ultraloq magnet as installation guide and test again？

**Comment by God-TM** [comment]
I'm not the one having issues with this lock (I figured I'd take pictures of a working environment if it could help anyone)... I'd rather not break what's currently working for me.

**Comment by JPDMaker** [comment]
I tested the lock jam bug. Does not update status. 

lock serial number: 8030305122130050


![IMG_8226](https://github.com/zwave-js/node-zwave-js/assets/32624354/a4339fcc-83ce-422e-a96d-dac1dc47230d)
![IMG_8228](https://github.com/zwave-js/node-zwave-js/assets/32624354/884fbc3b-71cb-44a0-a79d-bc5164945510)

This occurs both when the door is sensed to be closed or open 

![IMG_8231](https://github.com/zwave-js/node-zwave-js/assets/32624354/ad22d42e-0307-46bc-8f41-aab171566052)
![IMG_8230](https://github.com/zwave-js/node-zwave-js/assets/32624354/85b43a05-a2a5-4914-a064-fd3bb5144642)
Paper used to force jam while closed.

**Comment by jack0li** [comment]
Do our best to optimizing them.

**Comment by jack0li** [comment]
hi JPDMaker，
   Where should the status of Jam be displayed  in home assistance UI?

**Comment by JPDMaker** [comment]
It’s the state of the lock entity. Should update state to either ; Jammed, Locked, Locking, Unlocked, Unlocking, Unavailable, Unknown

[https://developers.home-assistant.io/docs/core/entity/lock/](https://developers.home-assistant.io/docs/core/entity/lock/)

**Comment by kpine** [comment]
> It’s the state of the lock entity. Should update state to either ; Jammed, Locked, Locking, Unlocked, Unlocking, Unavailable, Unknown

The Z-Wave integration does not support any of the extra lock states like jammed. It only supports locked or unlocked.

You can make HA feature requests at https://community.home-assistant.io/c/feature-requests/13.

**Comment by HarlemSquirrel** [comment]
Oh we definitely should have those states!

**Comment by HarlemSquirrel** [comment]
Took me a while to track this down but this appears to be set in https://github.com/home-assistant-libs/zwave-js-server-python/blob/0.51.1/zwave_js_server/const/command_class/lock.py#L81-L86

Then mapped in HA core in https://github.com/home-assistant/core/blob/2023.8.4/homeassistant/components/zwave_js/lock.py#L40-L49

**Comment by AlCalzone** [comment]
Jammed is not a lock state in `node-zwave-js`. It's a notification variable, which gets translated to a sensor in HA, I think here:
https://github.com/home-assistant/core/blob/7548c4aced3872c0d685ae0b07529a1f20fddbcd/homeassistant/components/zwave_js/discovery.py#L773-L785

**Comment by kpine** [comment]
Yes, there is a lock jammed sensor (https://github.com/home-assistant/core/blob/7548c4aced3872c0d685ae0b07529a1f20fddbcd/homeassistant/components/zwave_js/binary_sensor.py#L139-L145).

So the lock entity could use this value. PRs are welcome.

(Deleted my other incorrect reply).

**Comment by jack0li** [comment]
This issue( When the lock is jammed,the lock status is lock, user can not lock again) is fixed. We are trying to add the jammed notifacation funcation, need more time.

**Comment by itsensellc** [comment]
New Z-Wave U-Bolt Pro, latest firmware applied, latest HAOS/core, status is still stuck at open for u_bolt_pro_z_wave_smart_lock_current_status_of_the_door. The app shows correctly so the position sensor is working.

**Comment by AlCalzone** [comment]
I still find it tricky to trigger changes of the sensor without mounting the lock in a doorframe, but I haven't managed to get the state in the app and HA out of sync. When the status does change, I see it in both. Might be a connectivity issue?

**Comment by itsensellc** [comment]
> I still find it tricky to trigger changes of the sensor without mounting the lock in a doorframe, but I haven't managed to get the state in the app and HA out of sync. When the status does change, I see it in both. Might be a connectivity issue?

Don't think so? After latest update the door status seems to be working slightly better. What consistently seems to not work is auto-lock after X seconds doesn't seem to update the lock status.  Opening the door from outside and then it locking automatically when closed is working. But if I open it from HA, then it auto locks after 15 seconds... the status does not get updated.

**Comment by AlCalzone** [comment]
> auto-lock after X seconds doesn't seem to update the lock status

Wait, are we talking about the lock status or the door status?

**Comment by itsensellc** [comment]
> > auto-lock after X seconds doesn't seem to update the lock status
> 
> Wait, are we talking about the lock status or the door status?

Sorry I kind of switched gears. Door open status seems to be a bit better since the latest update. However the auto-lock has been an issue since adding the lock. Anytime I manually unlock it, that gets tracked properly in HA. Anytime it automatically locks it seems to not get tracked. I hit the lock in HA, and the status changes but the lock does nothing which is what I'd expect. So it knows it's already locked at least!

**Comment by AlCalzone** [comment]
Ok I'll try to reproduce that tomorrow

**Comment by jack0li** [comment]
I have tested in our site. The lock updates the correct status after auto-lock.   For testing door status,please move the lock body,don't move magnet.

The version of my lock.
The firmware version(in u-tec app show):02.26.0022
The Z-wave firmware version(HA show):1.5.3

Can you share me the testing video? We want more operate details to debug the issue. Thanks.

![auto-lock](https://github.com/zwave-js/node-zwave-js/assets/102127566/4ee93de4-d05f-4bf0-9d5d-4a86234d3e97)

**Comment by itsensellc** [comment]
> I have tested in our site. The lock updates the correct status after auto-lock. For testing door status,please move the lock body,don't move magnet.
> 
> The version of my lock. The firmware version(in u-tec app show):02.26.0022 The Z-wave firmware version(HA show):1.5.3
> 
> Can you share me the testing video? We want more operate details to debug the issue. Thanks.
> 
> ![auto-lock](https://user-images.githubusercontent.com/102127566/275678267-4ee93de4-d05f-4bf0-9d5d-4a86234d3e97.png)

Ok I have some further testing results. Door open status seems to be working perfectly fine for the moment. Door lock status status when opened at the lock seems perfectly fine, including when it auto locks. When I open the lock via the app, and it auto locks, status is reported fine. It's only when I open the door via HA that this issue occurs. And it doesn't seem to matter which kind of card I use. I can open the lock, it will update the status, but when the lock auto locks the HA state still reflects unlocked. When I click it again, it will beep and indicate lock. Sometimes this can get totally out of sync where it think it's unlocked but it's actually locked or vice versa. So it's only when controlling via HA that this is occurring. I can try and make a video if it's helpful I just wanted to clarify that first.

**Comment by jack0li** [comment]
Can you give me a interface picture that you operate the lock and the status show ? 
I will check it.

**Comment by itsensellc** [comment]
> Can you give me a interface picture that you operate the lock and the status show ? I will check it.

I will try and upload something a bit later. I did want to add I found something else interesting today. I opened the lock from inside to go out. It captured the door and lock sensors fine. However, it did not capture either door unlock/lock or open/closed events when I came back, using the fingerprint sensor and the door automatically locking when I shut it. I'm not sure if maybe it just didn't have time to register those events or they didn't get sent or what.

**Comment by jack0li** [comment]
My testing enviroment is:  a  Iaptop ( installed  HA virtual machine, Connected a SLUSB700A  hub with new driver, edge Brower  
 to login to HA in virtual machine and do operation  ); a u-bolt pro z-wave and magnet on a door. I don't use mobile APP.
what's your testing enviroment?  Where do you want to see the events? Let's keep in the same page.

**Comment by itsensellc** [comment]
> My testing enviroment is: a Iaptop ( installed HA virtual machine, Connected a SLUSB700A hub with new driver, edge Brower to login to HA in virtual machine and do operation ); a u-bolt pro z-wave and magnet on a door. I don't use mobile APP. what's your testing enviroment? Where do you want to see the events? Let's keep in the same page.

Ya not trying to muddy the waters. Just let me know what helps most to troubleshoot.

I'm running a dedicated ASRock BeeBox, with HAOS running Virtualbox (latest version, extension manager installed and USB passthrough), on Windows 10, Zooz ZST10-700 with firmware 7.19.2. Doesn't matter if I login with the HA web interface, or the HA app., I get the same result. I have the magnet on the door frame as well.

I am looking at the entity values/history directly for the most part. I know if those don't work then anything relying on them most likely won't either.

**Comment by jack0li** [comment]
Sorry, I just want to know how to reproduce the bug you mentioned in my side and try  help debug. I will update If i have new info.

**Comment by itsensellc** [comment]
100%! No need to apologize I appreciate your help. Will get you info soon.

**Comment by creeve4** [comment]
Any update on the last remaining bug?

**Comment by itsensellc** [comment]
> Any update on the last remaining bug?

Hi Chris - apologies I was travelling for the better part of the last month and a half. I will shoot a video for you and post it. It is still happening for sure.

**Comment by itsensellc** [comment]
Here's a quick video to illustrate what is happening: https://youtu.be/cmq1fcqTxyA

In short: if I click the unlock button, the lock unlocks as expected. Sometimes it takes a second to register on the card in HA but that works pretty consistently. But notice once the door lock auto locks (10 seconds after the unlock), the status doesn't update in HA. It remains as "unlocked". If I click "lock" again in HA you'll notice you hear one beep and the status updates in HA. I guess the lock knows it's locked hence the single beep. For whatever reason HA just won't update with that manual action. This only happens when I use the unlock button on one of the cards. If I use the Utec app, the status updates for both when it unlocks and auto locks as expected.

**Comment by holblin** [comment]
I agree, I have the same behavior in my instance.

**Comment by creeve4** [comment]
Thank you for sharing the video, I am experiencing the same with my lock.

**Comment by zirdum** [comment]
Interesting. Mine isn't doing that.

I click the unlock button, it unlocks. Mine is set to auto lock after 30 seconds. It locks and the status in HA updates immediately.

Interesting. If I set it to lock after 10 seconds, it does what you're experiencing. Setting it to 20 seconds works fine.

Hopefully this helps with troubleshooting the bug.

**Comment by itsensellc** [comment]
> Interesting. Mine isn't doing that.
> 
> I click the unlock button, it unlocks. Mine is set to auto lock after 30 seconds. It locks and the status in HA updates immediately.
> 
> Interesting. If I set it to lock after 10 seconds, it does what you're experiencing. Setting it to 20 seconds works fine.
> 
> Hopefully this helps with troubleshooting the bug.

I was optimistic when I saw this but I'm still having mixed results. I set my timeout to 20 seconds. The first time I tried the HA unlock, it worked as you described above. So I tried it again in succession, and it didn't work. Same result as I was getting before. After about 30 minutes I tried again, this time it worked. Again, tried in succession, it didn't work. Using the app, it works flawlessly. I can unlock, auto-lock, then unlock again in succession and the status is updated in HA as expected.

I will say I know when HA is not going to work because when I unlock the lock, it takes a few seconds for it to register in HA. When that happens, I know when it auto locks it's not going to update. When I unlock and it updates almost immediately, I know the auto lock status update is going to work.

**Comment by Daguse** [comment]
So HA removed the "Works With HA". Does that mean we are screwed?  Just hope that U-Tec will fix it at some point?

**Comment by balloob** [comment]
They fixed the issue and Home Assistant will offer users to update this firmware

**Comment by alaska99** [comment]
> They fixed the issue and Home Assistant will offer users to update this firmware

OK and where it is?? I still don't see firmware update.

**Comment by keck9939** [comment]
I find conflicting information on whether ULtraLoq is or is not a WWHA product. At this link: https://www.home-assistant.io/integrations/ultraloq/ it seems that it is, but I've seen threads that indicate ULtraLoq has been unresponsive to fixing issues and it has been removed from WWHA. I am considering replacing my current WiFi UltraLoq with the Z-Wave enabled version, but I don't want to do this if it is not reliably supported in HA. Can anyone shed some light on the state of UltraLoq support in HA?

**Comment by Daguse** [comment]
Late last month I did find a post on the Ultraloq forums talking about the same issue. In it someone from Ultraloq mentioned that a ticket would need to be opened and they would need to replace the lock. After opening the ticket and talking with someone they gave me this [LINK](https://file.u-tec.com/homekit/U-BOLT-PRO-ZWAVE_V1.7.gbl) to update the firmware from HA. That was about 2 weeks ago and things seem to be fixed.  

https://file.u-tec.com/homekit/U-BOLT-PRO-ZWAVE_V1.7.gbl

TLDR. Use the above link from Ultraloq and give it a try.

**Comment by grantland** [comment]
@Daguse how'd you install the firmware? I'm using HA with Z-Wave JS UI and I'm not sure how to upgrade with a downloaded firmware file. Both HA and Z-Wave JS UI shows the device is up-to-date with no option to provide my own firmware files.

**Comment by zirdum** [comment]
What firmware does it show? Mine says it's v1.5.3 which I assume is current for the z-wave model. I do see on their site that 1.41 is available, but it seems only for the wi-fi models.

**Comment by grantland** [comment]
@zirdum in HA/Z-Wave-JS it shows 1.5.3, however, in U home it shows 02.26.0022. The FW link above seems to indicate it's 1.7

**Comment by rayarell** [comment]
> Late last month I did find a post on the Ultraloq forums talking about the same issue. In it someone from Ultraloq mentioned that a ticket would need to be opened and they would need to replace the lock. After opening the ticket and talking with someone they gave me this [LINK](https://file.u-tec.com/homekit/U-BOLT-PRO-ZWAVE_V1.7.gbl) to update the firmware from HA. That was about 2 weeks ago and things seem to be fixed.
> 
> https://file.u-tec.com/homekit/U-BOLT-PRO-ZWAVE_V1.7.gbl
> 
> TLDR. Use the above link from Ultraloq and give it a try.

Thank you for posting this! It fixed my issues!
